ext/spl/tests/bug74669.phpt

```diff
@@ -105,6 +105,8 @@ object(SelfArray)#9 (1) {
   string(3) "bar"
 }
 string(77) "O:9:"SelfArray":4:{i:0;i:16777216;i:1;N;i:2;a:1:{s:3:"foo";s:3:"bar";}i:3;N;}"
+
+Deprecated: Creation of dynamic property SelfArray::$foo is deprecated in %s on line %d
```

Comment:
I think that this deprecation notice is probably expected, since it really is creating foo in the object's hash table

`spl_array_write_dimension_ex` probably needs to be changed to start warning about creating a deprecated dynamic property

```php
php > class MyClass {}
php > $c = new MyClass();
php > $y = new ArrayObject($c);
php > $y['foo'] = 'bar'; // <----- should warn about creating dynamic property MyClass. Same for integer offset.
php > $c->foo2 = 2;

Deprecated: Creation of dynamic property MyClass::$foo2 is deprecated in php shell code on line 1
php > var_export($c);
\MyClass::__set_state(array(
   'foo' => 'bar',
   'foo2' => 2,
))
```

Reply:
> `spl_array_write_dimension_ex` probably needs to be changed to start warning about creating a deprecated dynamic property

It's fine with me to treat this as an entirely different bug and fixing it elsewhere, maybe adding a todo comment referencing the github issue. I reported two issues while looking into this - #9432 affects php 8.2 dynamic property deprecation and `readonly` (and maybe some internal classes such as pecls, haven't looked)

#9434 affects php 7.4+